exposed-java-time/src/test/kotlin/org/jetbrains/exposed/JavaTimeTests.kt

```diff
@@ -442,25 +451,30 @@ fun <T : Temporal> assertEqualDateTime(d1: T?, d2: T?) {
 }
 
 private fun assertEqualFractionalPart(nano1: Int, nano2: Int) {
-    when (currentDialectTest) {
-        // nanoseconds (H2, Oracle & Sqlite could be here)
-        // assertEquals(nano1, nano2, "Failed on nano ${currentDialectTest.name}")
+    val dialect = currentDialectTest
+    val db = dialect.name
+    when (dialect) {
         // accurate to 100 nanoseconds
-        is SQLServerDialect -> assertEquals(roundTo100Nanos(nano1), roundTo100Nanos(nano2), "Failed on 1/10th microseconds ${currentDialectTest.name}")
+        is SQLServerDialect ->
+            assertEquals(roundTo100Nanos(nano1), roundTo100Nanos(nano2), "Failed on 1/10th microseconds $db")
         // microseconds
-        is H2Dialect, is MariaDBDialect, is PostgreSQLDialect, is PostgreSQLNGDialect ->
-            assertEquals(roundToMicro(nano1), roundToMicro(nano2), "Failed on microseconds ${currentDialectTest.name}")
+        is H2Dialect, is PostgreSQLDialect ->
+            assertEquals(roundToMicro(nano1), roundToMicro(nano2), "Failed on microseconds $db")
+        is MariaDBDialect ->
+            assertEquals(floorToMicro(nano1), floorToMicro(nano2), "Failed on microseconds $db")
         is MysqlDialect ->
-            if ((currentDialectTest as? MysqlDialect)?.isFractionDateTimeSupported() == true) {
+            if ((dialect as? MysqlDialect)?.isFractionDateTimeSupported() == true) {
                 // this should be uncommented, but mysql has different microseconds between save & read
 //                assertEquals(roundToMicro(nano1), roundToMicro(nano2), "Failed on microseconds ${currentDialectTest.name}")
             } else {
                 // don't compare fractional part
             }
```

Comment:
I'm not sure why this block is doing nothing and what that means for MySQL tests that rely on this assertion, so it's been added to my list of things to investigate.

Reply:
let's rewrite it